proposals/0475-tuple-syntax.rst

```diff
@@ -63,15 +63,27 @@ variable in a bit of code. This allows us to e.g. refer to ``Tuple2`` and ``Tupl
 Proposed Change Specification
 -----------------------------
 
-1. Create a new module in ``base`` called ``GHC.Prelude``. This module exports definitions
-   peculiar to GHC that are safe to use in ordinary code.
-   This module is *not* intended as a replacement to ``Prelude``,
-   but instead a complement to it. This module is a counterpart to ``GHC.Exts``, which exports
+1. Create three new modules in ``ghc-experimental`` called ``GHC.Prelude``,
+   ``GHC.Tuple`` and ``GHC.Sum``.
+   Since this new paradigm may evolve in response to users' experience, we want
+   to avoid committing to an API in ``base`` too early, and therefore choose the
+   new package as the home for the exports of the new entities.
+
+   ``GHC.Prelude`` exports definitions peculiar to GHC that are safe to use in
+   ordinary code, but whose API may evolve rapidly over subsequent releases.
+   This module is *not* intended as a replacement to ``Prelude``, but instead a
+   complement to it. This module is a counterpart to ``GHC.Exts``, which exports
    many *unsafe* internals.
 
    This proposal lists several definitions to be exported from ``GHC.Prelude``, but leaves other
    definitions to be added in separate proposal(s).
 
+   ``GHC.Tuple`` will clash with the same module from ``ghc-prim``.
```

Comment:
Can we avoid the clashes? Nothing compels us to create a conflicting `GHC.Tuple` in `ghc-experimental`, at least not for the sake of this proposal. Just leave it in `ghc-prim`?

The new `GHC.Sum` and `GHC.Prelude` can go to `ghc-experimental`, of course, as there is no conflict.

Reply:
> Can we avoid the clashes? Nothing compels us to create a conflicting GHC.Tuple in ghc-experimental, at least not for the sake of this proposal. Just leave it in ghc-prim?

Good qn.  But if we leave it in `ghc-prim` then clients of the new tuple stuff will be force to depend on `ghc-prim` which we do not want.  Rather, I think we should transition `ghc-prim:GHC.Tuple` to become `ghc-prim:GHC.Internals.Tuple`.  Quite how fast we do that isn't clear to me.  

@goldfirere ?

Reply:
our new `GHC.Tuple` depends on `base` (it needs `TypeNats` and possible `TypeError`), so leaving it in `ghc-prim` is a no-go.